### ISSUE TITLE
[Account] Multiple API Keys for one account

### ISSUE THREAD

**Issue by marm123** [serpapi/public-roadmap#270]
We have reoccurring requests to make it possible to have multiple API Keys assigned to a single account.
We can have this set up for bigger plans like Infrastructure Plan and higher.
Additionally, we may add a field for each API Key where users can name them according to their needs so they are not confused.
It can look something like below:
<img width="748" alt="api_keys" src="https://user-images.githubusercontent.com/28161162/185907031-610cfa19-9c12-450a-afd3-3091d4dd67d9.png">

<a href="https://serpapi.com/manage-api-key">SerpApi - Manage API Key</a>

**Comment by ilyazub** [comment]
@marm123 Previously customers have asked about multiple API keys to monitor their usage. Searches and credits are bound to the account.

What if we'll add a query parameter like `serpapi_project_id` and add it to the response as `search_metadata.serpapi_project_id`? This way we won't increase the complexity of authentication.

**Comment by ritu1337** [comment]
Could also implement sub-accounts so that every sub-account (sub-user) could have a different API key.

**Comment by hartator** [comment]
> Could also implement sub-accounts so that every sub-account (sub-user) could have a different API key.

Yes, just allowing an account to have sub-accounts might be an easier way to allow multiple API keys.

As we don't have to reinvent our current `user.api_key` and rework its indexes. And maybe we can have a naive implementation where we don't have roles yet. i.e., every account can do everything (billing, stats, search history) except they have each a different API key?

**Comment by schaferyan** [comment]
This continues to be one of our most commonly requested features. A lot of users specifically request an "Organization Account" with sub-accounts.

**Comment by hartator** [comment]
Maybe let's target to have a simple implementation of this deployed?

I.e., root users can now invite subusers. Subusers have the same access, same billing, and same plan as root user. Subusers can do anything as the root user can do except removing the root user. Subusers and root user have all different API key while still sharing root user plan, monthly searches, and credits. Subusers and root user can all view all aggregated usage (searches list and stats) or see usage for specific user (i.e., specific API key).

This is way it will cover the use case of a business trying to invite their team over to collaborate on serpapi.com and also the use case of a business trying to have different account with different functions (billing, management, engineering).

**Comment by marm123** [comment]
Two customers requested this feature:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000042185)
[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000042182)

**Comment by schaferyan** [comment]
Another customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000042292?view=List)

**Comment by marm123** [comment]
Customer is looking to scale their business significantly, but they require this feature or a workaround:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000042185?view=List)

**Comment by marm123** [comment]
Another user requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000045168?view=List)

**Comment by hartator** [comment]
Closed in favor of https://github.com/serpapi/public-roadmap/issues/1017.

**Comment by sonika-serpapi** [comment]
A customer mentioned they would like an official way to set different API keys for development, staging, production, etc. They were looking for a better way to do this than using the team feature. 
[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000077455?view=List)

**Comment by schaferyan** [comment]
@sonika-serpapi I think this would make more sense as a new issue, since we already have support for Multiple API keys per account. Can you also ask them for more details on what is missing from the current implementation for them? As it is indeed already possible to set different API keys for use with different environments.

**Comment by sonika-serpapi** [comment]
Sounds good, I've created a new issue here: https://github.com/serpapi/public-roadmap/issues/2527
I've also noted more details there.
